### DISCUSSION TITLE
How to filter write message that don't have "written"?

### DISCUSSION THREAD

**Discussion by cbochs** [folke/noice.nvim#364] [Q&A] [answered]
I'm trying to filter out "write" messages. I've [read the wiki](https://github.com/folke/noice.nvim/wiki/Configuration-Recipes#hide-written-messages-1) and extended LazyVim with the suggested config options:

```lua
{
    "folke/noice.nvim",
    opts = {
        routes = {
            {
                filter = {
                    event = "msg_show",
                    kind = "",
                    any = {
                        { find = "written" },
                    },
                },
                opts = { skip = true },
            },
        },
    },
},
```

However, I'm still getting notify messages when I type `:w`

<img width="502" alt="image" src="https://user-images.githubusercontent.com/2467016/219937036-111407eb-a0ca-4728-a35f-4a561b465b16.png">

I've looked at `:Noice` and see the following logs:

```
01:20:47 msg_show > w "nvim/lua/plugins/init.lua" 
01:20:47 msg_show > w "nvim/lua/plugins/init.lua" 108L, 3567B
```

I tried a couple alternative route filters without success:

* `find = "^w"` or `find = "w"`
* `find = "%dL, %dB"`

I'm just wondering, what pattern should be used to filter out these kind of messages?

Thanks!

**Comment by folke** [comment]
https://github.com/folke/noice.nvim/wiki/A-Guide-to-Messages#handling-hit-enter-messages
Easier to just unset that written option in shortmess

**Comment by cbochs** [comment]
Great! I'll make sure to thoroughly read the Guide to Messages. Thanks you for the tip 🙂

**Comment by cbochs** [comment]
Quick note: just checked the `shortmess`. This is what I have:

```lua
print(vim.o.shortmess)
-- Prints: iltIWcsnfTxOFoC
```

I also made sure "W" was set in the shortmess:

```lua
vim.opt.shortmess:append("W")
```

And from what I can tell, this _should_ disable write messages(?):
 
```
'shortmess' 'shm'	string	(default "filnxtToOF")
			global
	This option helps to avoid all the |hit-enter| prompts caused by file
	messages, for example  with CTRL-G, and to avoid some other messages.
	It is a list of flags:
	 flag	meaning when present	 
	  ...
	  W	don't give "written" or "[w]" when writing a file	*shm-W*
```

I'm still seeing the notify messages popup:

<img width="470" alt="image" src="https://user-images.githubusercontent.com/2467016/219938387-72b4ccb8-9a0b-40dc-ae8c-cc4dfd0670c5.png">

**Reply by folke** [reply]
no idea. You'll have to figure this out yourself. Noice is just relaying messages. You are responsible for configuring Neovim's messages. Not Noice.

**Reply by cbochs** [reply]
Sound good, will keep investigating. I've tried a catch-all route and it seems to work well enough for now:

```lua
filter = {
    event = "msg_show",
    kind = "",
    opts = { skip = true },
}
```

Thanks

**Comment by cbochs** [comment, marked as answer]
@folke coming back to this,

I've remapped `;` as `:` so my workflow for saving is typically just a two keystroke `:w`. It appears that even when appending the `W` shortmess option, write messages will continue to appear unless explicitly silenced (e.g. `:silent w`). This is what I came up with to filter write messages:

```lua
    { -- add message filters
        "folke/noice.nvim",
        opts = function(_, opts)
            opts.routes = {
                { -- filter write messages "xxxL, xxxB"
                    filter = {
                        event = "msg_show",
                        find = "%dL",
                    },
                    opts = { skip = true },
                },
            }
        end,
    },

```
